### ISSUE TITLE
aurelia-ace experiments

### ISSUE THREAD

**Issue by adriatic** [aurelia-tools/act#2]
### simplest new project

```
What would you like to name your new project?

> new-project

Would you like to use the default setup or customize your choices?

1. Default ESNext (Default)
   A basic web-oriented setup with Babel for modern JavaScript development.
2. Default TypeScript
   A basic web-oriented setup with TypeScript for modern JavaScript development.
3. Custom
   Select transpilers, CSS pre-processors and more.

> 1

Project Configuration

    Name: new-project
    Platform: Default
    Transpiler: Babel
    CSS Processor: None
    Unit Test Runner: Karma
    Editor: Visual Studio Code


Would you like to create this project?

1. Yes (Default)
   Creates the project structure based on your selections.
2. Restart
   Restarts the wizard, allowing you to make different selections.
3. Abort
   Aborts the new project wizard.

> 1
Project structure created and configured.

Would you like to install the project dependencies?

1. Yes (Default)
   Installs all server, client and tooling dependencies needed to build the
   project.
2. No
   Completes the new project wizard without installing dependencies.

> 1

Installing project dependencies.
```

___many minutes of waiting___

lots of output - see the attached file 

```
Congratulations! Your Project "new-project" Has Been Created!


Now it's time for you to get started. It's easy. First, change directory into your new
project's folder. You can use cd new-project to get there. Once in your
project folder, simply run your new app with au run. Your app will run
fully bundled. If you would like to have it auto-refresh whenever you make changes to your
HTML, JavaScript or CSS, simply use the --watch flag. If you want to build your app
for production, run au build --env prod. That's just about all there is to
it. If you need help, simply run au help.


Happy Coding!
install:new-project       \ |##########################################################----|
```

**Comment by Thanood** [comment]
Experimenting with it (1st pass :smile:) shows the majority of changes (transpiler and css processor choice) between configurations take place in `package.json` and `aurelia_project/aurelia.json` as well as `aurelia_project/tasks/(transpile|process-css).js`.

For transpilers, only a different configuration is set in `aurelia_project/aurelia.json`. For TypeScript all tasks are `.ts` files. Of course, `package.json` is either filled with `babel` or `typescript` dependencies and their `gulp` counterparts.

For css preprocessors the file `aurelia_project/tasks/(transpile|process-css).js` is changed to produce output using the chosen preprocessor with a setting changed in `aurelia_project/aurelia.json`.

Sample files in `src` are only changed to `.ts` files if you choose TypeScript.
Bundle configuration stays the same regardless of choices.

So in summary most changes are done in configuration and gulp tasks.

**Comment by adriatic** [comment]
I just **knew** that we need your systematic approach as well as creativity to get this going, so I would much prefer if you take this under your control, where I would pepper you with questions. 

Organizing the "generated data" as a [cli-outputs](https://github.com/Thanood/cli-outputs) is a terrific idea as it will help us manage all data without a loss. 

**my first question is this:**

How does cli create a file like **[app.js](https://github.com/Thanood/cli-outputs/blob/master/default/esnext/src/app.js)** (one of the simplest examples)? 

Our core team guys are all pros so, they must have done something similar to what every compiler does - code generation phase, that follows the parsing phase. Our situation is similar in the sense that cli picks the data from the user and then goes on and spits out both application source files and all needed configuration files. 

You focused on configuration files first (a very good decision) and I would like you to explain to me how is the source code generated.

**Comment by Thanood** [comment]
Short answer for now: it's taken from here:  https://github.com/aurelia/cli/tree/master/lib/resources/src

I will have a deeper look at the scaffolding process later.

**Comment by adriatic** [comment]
Are you saying that cli has a repo for each of its "outcome case" and then copies the content of that repo into the "output stream". 

While this is probably the most efficient way, it is also the most primitive :-)
